### PR TITLE
Worker connector updates & refactoring

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description

Updated the `WorkerConnector` implementations with these https://github.com/spatialos/gdk-for-unity/pull/1365 changes. Additionally made all the map building use `Task` rather than coroutines and did some refactoring. 

#### Tests

- [x] Ran it locally
- [x] Release QA pipeline
- [ ] Test built out client

#### Documentation

- [x] Changelog

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/WorkerConnectors/SimulatedPlayerCoordinatorWorkerConnector.cs [hunk 1]
```diff
@@ -30,6 +30,8 @@ public class SimulatedPlayerCoordinatorWorkerConnector : WorkerConnectorBase
         public int MaxSimulatedPlayerCount = 1;
         public int SimulatedPlayerCreationInterval = 5;
 
+        public Bounds Bounds { get; set; }
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/WorkerConnectors/SimulatedPlayerCoordinatorWorkerConnector.cs:33]: private set?

File: workers/unity/Assets/Fps/Scripts/WorkerConnectors/WorkerConnectorBase.cs [hunk 2]
```diff
@@ -44,34 +27,42 @@ public override void Dispose()
...
-                mapTemplate,
-                worldSize,
-                transform,
-                Worker.WorkerType,
-                this);
+            LevelInstance = await MapBuilder.GenerateMap(mapTemplate, worldSize, transform, Worker.WorkerType);
         }
 
-        protected int GetWorldSize()
+        protected async Task WaitForWorkerFlags(CancellationToken token, params string[] flagKeys)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/WorkerConnectors/WorkerConnectorBase.cs:42]: This looks like it is worth integrating into the core later?

File: workers/unity/Assets/Fps/Scripts/WorkerConnectors/SimulatedPlayerCoordinatorWorkerConnector.cs (hunk 1 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Fps/Scripts/WorkerConnectors/SimulatedPlayerCoordinatorWorkerConnector.cs:33]: Ah yes - I meant to make it private set 😅

**Comment by zeroZshadow** [comment]
Random through, does the map generator still work in the editor with this? Not when playing.

**Comment by jamiebrynes7** [comment]
> Random through, does the map generator still work in the editor with this? Not when playing.

Yup - just tested with a map w/ 200 tile layers, Editor pauses for a few seconds and then completes fine

**Comment by jamiebrynes7** [comment]
Everything looked good on the quick playtest! 📈

**Pull request merged**